crates/tick_probe/src/main.rs

```diff
@@ -22,12 +22,13 @@ fn main() -> Result<()> {
         None => Client::try_default(),
     }?;
 
-    let event = Event::TestTick;
+    let event = Event::new(EventKind::TestTick);
+    let message = EventMessage::with_event(event);
```

Comment:
nit: when writing initialization chains such as this I try to make it clear which is the final variable that will be used later and which are just the transient ones. In this case, I'd just collapse everything into a single line:

```suggestion
    let message = EventMessage::with_event(Event::new(EventKind::TestTick));
```

when the chain is longer I use a block, like this:

```suggestion

    let message = {
        let event = Event::new(EventKind::TestTick);
        EventMessage::with_event(event);
     }
```

so that it's clear `event` has no use beyond here.

Reply:
> In this case, I'd just collapse everything into a single line: (...)

I might also instinctively write it as as such, looking at it next to Ryo's original version I find Ryo's one more readable and clear. No need to visually parse where the parens are; and giving a name to a value can sometimes do a lot of service to clarity. (some our internal review comments are "can you please name this value?").

> when the chain is longer I use a block, like this: (...)

Yeah that does look somewhat cleaner here locally. But if we were to consistently apply such a rule I think we would have a _lot_ of curly braces and extra indents in the codebase, negatively impacting readability in the end.

By the way, at least to rust compiler, it is clear that `event` is not going to be used after line 26: `struct Event` is not `Copy` and `EventMessage::with_event(event)` _consumes_ the event. I.e. using it below should not borrow-check.

Not sure if "think about it like rustc does" is a valid point on a code review though. :upside_down_face:

Reply:
> Yeah that does look somewhat cleaner here locally. But if we were to consistently apply such a rule I think we would have a lot of curly braces and extra indents in the codebase, negatively impacting readability in the end.

Agree. I didn't meant this to be a *rule* - just wanted to share how I tend to write things. These blocks IMO make sense only when the initialization has more than 2-3 transitive variables.

> Not sure if "think about it like rustc does" is a valid point on a code review though. 🙃

I think it is - if I was familiar enough with the borrow-checking rules I probably wouldn't even consider writing this remark.

When writing/reviewing code we always have to assume some level of reader's familiarity with the language. In this case I think it's fine to assume familiarity with how lifetimes work.

I'll resolve this. As said, no strong opinion here - just wanted to share alternatives. .)